### ISSUE TITLE
Fedora 34 / Gnome Kooha not working

### ISSUE THREAD

**Issue by kronikpillow** [SeaDve/Kooha#72]
so, I'v recently switched to Fedora 34 to try out wayland and the new Gnome 40, I do some youtube content related to a browser game, and I was looking for a screen recorder for wayland and found Kooha, I installed it via flatpak, followed the install instructions from this github page, and used the command to set it to 60 FPs, when I try to record my screen Kooha completely freezes, then it tells me that Kooha is not responding, asks me to Force to quit or Wait, I opt for Wait, same thing happens after waiting 2-3 times, and then it starts recording my screen, but after it starts recording my screen, and I finish the recording, i click Stop recording, Kooha again freezes, then unfreezes, and continues to record my screen, I literally can't stop the recording ... anyone experiencing similar issues? would be great for any input/guidance on if im missing something or what i could be doing wrong, if you need some logs or anything relevant please ask, I will provide what ever info you need

for starters, i'm on a AMDGPU (AMD Radeon R9 380 4GB Graphics card)

**Comment by SeaDve** [comment]
I think it might be the 60 fps. One of the reason why I didn't put it to ui is that it is buggy, even for me.

Can you try it with default settings (30 fps)? It may be a different issue, other than the fps

**Comment by kronikpillow** [comment]
just tried it, same thing happens ...

another thing it wont let me do is select the location to where to record the video, when i open that panel, and select the actual folder, the only button i have is cancel, no option to actually select it and accept the location

**Comment by SeaDve** [comment]
I'm currently working on a new backend on portal branch, I think it might fix the issue. If it's okay to you, you can build it from source and test it.

With the saving location one, may you elaborate that further?

**Comment by SeaDve** [comment]
https://github.com/SeaDve/Kooha/tree/portal

I recommend on using gnome builder to build it. Or use meson instructions, but you have to install dependencies

**Comment by heyahmed** [comment]
Having the same issue on Fedora 34. For me, the application and all its settings are set to default and I installed it from flathub. As soon as I click the record button, no matter what the configuration of recording is, the application hangs and I get a force quit window.

**Comment by SeaDve** [comment]
For some reason, I can't reproduce it despite of also being in Fedora 34. Which Kooha version are you?

**Comment by heyahmed** [comment]
It was the latest version on flathub, 1.2.1.
I installed it using the Gnome Builder method, but same thing. As soon as I click on record, it crashes.

This was in the terminal at that time, don't know what it means.
`(io.github.seadve.Kooha:2): Gtk-CRITICAL **: 17:45:27.823: Unable to connect to the accessibility bus at 'unix:abstract=00008': Could not connect: Connection refused
`

**Comment by SeaDve** [comment]
In what branch did you build kooha? you could do `git fetch origin` then `git checkout portal` then rebuild

**Comment by heyahmed** [comment]
After doing that, it worked. I could record my full screen, also record a section of the screen. The sound and mouse pointer were also working.

**Comment by SeaDve** [comment]
Did you saw this window?
![image](https://user-images.githubusercontent.com/64297935/121359368-01dc6400-c966-11eb-8a6d-ed28a50feb0a.png)

**Comment by heyahmed** [comment]
Yes.

**Comment by SeaDve** [comment]
Okay, thanks! That's a good sign, the fix will arrive on 2.0.0

**Comment by garrett** [comment]
Note: There was a bug in PipeWire 0.3.29 that caused Kooha to not work (among a few other issues). The issue was fixed in PipeWire 0.3.30. Please make sure you have the latest version of PipeWire in Fedora and try again.

(I also had an issue with Kooha where it would freeze last week. It worked fine with previous versions of PipeWire and works fine again for me with 0.3.30.)

**Comment by SeaDve** [comment]
thanks, so this seems to be not kooha

**Comment by atombrenner** [comment]
I'm using Fedora 34 with Wayland, Kooha 1.2.1 and pipewire 0.3.32. Works like a charm ;-)

**Comment by garrett** [comment]
@atombrenner: Right, same here. It works on my system again too.

It was a bug in a specific version of PipeWire, as I mentioned above. It affected more than just Kooha and was resolved somewhat quickly. Anyone using an up-to-date Fedora for that short time hit the bug with Kooha (and other apps).

We didn't get confirmation from the original reporter, but I did hit the same problem at the same time and tracked down the issue and confirmed it was fixed with the newer version, so this issue should probably be closed.
